test/common/EdgeConfiguration.cs

```diff
@@ -0,0 +1,216 @@
+        {
+            this.config = GetBaseConfig(agentImage);
+            this.iotHub = iotHub;
+            this.deviceId = deviceId;
+        }
+
+        public void AddRegistryCredentials(string address, string username, string password)
+        {
+            this.UpdateAgentDesiredProperties(desired =>
+            {
+                JObject settings = desired.Get<JObject>("runtime").Get<JObject>("settings");
+                settings.Add("registryCredentials", JToken.FromObject(new
+                {
+                    reg1 = new
+                    {
+                        username,
+                        password,
+                        address
+                    }
+                }));
+            });
+        }
+
+        // Adds proxy information to each module in Edge Agent's desired properties. Call this
+        // method after you've added all the modules that need proxy information.
+        public void AddProxy(Uri proxy)
+        {
+            this.ForEachModule((name, module) =>
+            {
+                JObject env = GetOrAddObject("env", module);
+                env.Add("https_proxy", JToken.FromObject(new
+                {
+                    value = proxy.ToString()
+            }));
+                env.Add("UpstreamProtocol", JToken.FromObject(new
+                {
+                    value = UpstreamProtocolType.AmqpWs.ToString()
+                }));
+            });
+        }
+
+        public void AddEdgeHub(string image)
+        {
+            this.UpdateAgentDesiredProperties(desired =>
+            {
+                JObject systemModules = desired.Get<JObject>("systemModules");
+                systemModules.Add("edgeHub", JToken.FromObject(new
+                {
+                    type = "docker",
+                    status = "running",
+                    restartPolicy = "always",
+                    settings = new
+                    {
+                        image,
+                        createOptions = "{\"HostConfig\":{\"PortBindings\":{\"8883/tcp\":[{\"HostPort\":\"8883\"}],\"443/tcp\":[{\"HostPort\":\"443\"}],\"5671/tcp\":[{\"HostPort\":\"5671\"}]}}}"
+                    }
+                }));
+            });
+
+            // { "modulesContent": { "$edgeHub": { ... } } }
+            this.config.ModulesContent["$edgeHub"] = new Dictionary<string, object>
+            {
+                ["properties.desired"] = new
+                {
+                    schemaVersion = "1.0",
+                    routes = new Dictionary<string, string>
+                    {
+                        ["route1"] = "from /* INTO $upstream",
+                    },
+                    storeAndForwardConfiguration = new
+                    {
+                        timeToLiveSecs = 7200
+                    }
+                }
+            };
+        }
+
+        public void AddTempSensor(string image)
```

Comment:
Suggest to make it more generic; so that we can add different modules.

Reply:
Yes I'm sure it will become more generic as I develop more tests.